cmd/snap-seccomp/main.go

```diff
@@ -790,27 +789,16 @@ func compile(content []byte, out string) error {
 	}
 
 	// write atomically
-	dir, err := os.Open(filepath.Dir(out))
-	if err != nil {
-		return err
-	}
-	defer dir.Close()
-
-	fout, err := os.Create(out + ".tmp")
+	fout, err := osutil.NewAtomicFile(out, 0644, 0, -1, -1)
 	if err != nil {
 		return err
 	}
 	defer fout.Close()
-	if err := secFilter.ExportBPF(fout); err != nil {
-		return err
-	}
-	if err := fout.Sync(); err != nil {
-		return err
-	}
-	if err := os.Rename(out+".tmp", out); err != nil {
+
+	if err := secFilter.ExportBPF(fout.BackingFile()); err != nil {
```

Comment:
The new method doesn't seem to be doing much good here. Why not using fout directly?

Reply:
Not that fout itself (or fout.File) _are_ the backing file.

Reply:
Note that fout itself (and fout.File) _are_ the backing file.

Reply:
The way osutil.NewAtomicFile() is written currently does not allow me to use it directly (or fout.File). The reason is that NewAtomicFile returns on interface of AtomicWriter not osutil.atomicFile. However that can be easily changed of course.

Details with the current code:
```
with just fout:
./main.go:798: cannot use fout (type osutil.AtomicWriter) as type *os.File in argument to secFilter.ExportBPF
with fout.File:
./main.go:798: fout.File undefined (type osutil.AtomicWriter has no field or method File)
```

I can of course tweak osutil/io.go and expose the osutil.AtomicFile itself, something like http://paste.ubuntu.com/25525999/ (plus updating all the tests).